webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+package com.box.l10n.mojito.service.scheduledjob.jobs;
+
+import com.box.l10n.mojito.entity.ScheduledJob;
+import com.box.l10n.mojito.pagerduty.PagerDutyException;
+import com.box.l10n.mojito.pagerduty.PagerDutyIntegrationService;
+import com.box.l10n.mojito.pagerduty.PagerDutyPayload;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.service.pollableTask.PollableFuture;
+import com.box.l10n.mojito.service.scheduledjob.IScheduledJob;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
+  private Long pollableTaskId;
+
+  @Override
+  public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    // Fetch the scheduled job and cast the properties
+    scheduledJob = scheduledJobRepository.findByJobKey(jobExecutionContext.getJobDetail().getKey());
```

Comment:
Should this not be the try as well? 
What if the cast fails or if the key is not found (by some misfortune)

Reply:
The only way for this not to exist if there was some manual altering of the database, the manager creates a job for reach scheduled job so it must exist otherwise it would never even get to this execute method.